File: chat-uikit/src/main/java/io/agora/chat/uikit/activities/EaseChatHistoryActivity.java
package io.agora.chat.uikit.activities;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.fragment.app.Fragment;

import io.agora.chat.ChatMessage;
import io.agora.chat.uikit.R;
import io.agora.chat.uikit.base.EaseBaseActivity;
import io.agora.chat.uikit.chathistory.EaseChatHistoryFragment;
import io.agora.chat.uikit.constants.EaseConstant;
import io.agora.chat.uikit.databinding.EaseActivityFragmentBinding;
import io.agora.chat.uikit.widget.EaseTitleBar;

public class EaseChatHistoryActivity extends EaseBaseActivity {
    private EaseActivityFragmentBinding binding;
    private ChatMessage combineMessage;

    public static void actionStart(Context context, ChatMessage combineMessage) {
        Intent intent = new Intent(context, EaseChatHistoryActivity.class);
        intent.putExtra(EaseConstant.EXTRA_CHAT_COMBINE_MESSAGE, combineMessage);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle arg0) {
        super.onCreate(arg0);
        binding = EaseActivityFragmentBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        mContext= this;
        initIntent(getIntent());
        initView();
    }

    public void initIntent(Intent intent) {
        combineMessage = intent.getParcelableExtra(EaseConstant.EXTRA_CHAT_COMBINE_MESSAGE);
    }

    public void initView() {
        Fragment fragment = getSupportFragmentManager().findFragmentByTag("chat_history");
        if(fragment == null) {
            EaseChatHistoryFragment.Builder builder = new EaseChatHistoryFragment.Builder(combineMessage)
                    .useHeader(true)
                    .enableHeaderPressBack(true)
                    .setHeaderBackPressListener(new EaseTitleBar.OnBackPressListener() {
                        @Override
                        public void onBackPress(View view) {
                            onBackPressed();
                        }
                    })
                    .setEmptyLayout(R.layout.ease_layout_no_data_show_nothing);
            fragment = builder.build();
        }
        getSupportFragmentManager().beginTransaction().replace(R.id.fl_fragment, fragment, "chat_history").commit();
    }
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/activities/EaseChatHistoryActivity.kt
package io.agora.uikit.feature.chat.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import io.agora.uikit.EaseIM
import io.agora.uikit.R
import io.agora.uikit.base.EaseBaseActivity
import io.agora.uikit.common.ChatMessage
import io.agora.uikit.common.EaseConstant
import io.agora.uikit.common.extensions.hasRoute
import io.agora.uikit.common.extensions.showToast
import io.agora.uikit.databinding.EaseActivityChatBinding
import io.agora.uikit.feature.chat.chathistory.EaseChatHistoryFragment
import io.agora.uikit.feature.chat.interfaces.OnCombineMessageDownloadCallback
import io.agora.uikit.feature.chat.interfaces.OnSendCombineMessageCallback

open class EaseChatHistoryActivity: EaseBaseActivity<EaseActivityChatBinding>() {
    override fun getViewBinding(inflater: LayoutInflater): EaseActivityChatBinding? {
        return EaseActivityChatBinding.inflate(inflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val message = intent.getParcelableExtra(EaseConstant.EXTRA_CHAT_COMBINE_MESSAGE) as? ChatMessage
        if (message == null) {
            mContext.showToast("Message is null.")
            finish()
            return
        }
        var fragment = supportFragmentManager.findFragmentByTag("ease_chat_history")
        if (fragment == null) {
            val builder = EaseChatHistoryFragment.Builder(message)
                .useTitleBar(true)
                .setTitleBarTitle(getString(R.string.ease_combine_default))
                .enableTitleBarPressBack(true)
                .setTitleBarBackPressListener { finish() }
                .setEmptyLayout(R.layout.ease_layout_no_data_show_nothing)
                .setOnSendCombineMessageCallback(object : OnSendCombineMessageCallback {

                    override fun onSendCombineError(
                        message: ChatMessage?,
                        code: Int,
                        errorMsg: String?
                    ) {
                        // Handle the error message
                    }
                })
                .setOnCombineMessageDownloadCallback(object: OnCombineMessageDownloadCallback {
                    override fun onDownloadSuccess(messages: List<ChatMessage>) {

                    }

                    override fun onDownloadFail(error: Int, errorMsg: String?) {
                        errorMsg?.apply {
                            mContext.showToast(this)
                        }
                    }
                })
            setChildSettings(builder)
            fragment = builder.build()
        }
        fragment?.let { fragment ->
            supportFragmentManager.beginTransaction().replace(binding.flFragment.id, fragment).commit()
        }
    }

    protected open fun setChildSettings(builder: EaseChatHistoryFragment.Builder) {}

    companion object {
        fun actionStart(context: Context, combineMessage: ChatMessage?) {
            Intent(context, EaseChatHistoryActivity::class.java).apply {
                putExtra(EaseConstant.EXTRA_CHAT_COMBINE_MESSAGE, combineMessage)
                EaseIM.getCustomActivityRoute()?.getActivityRoute(this.clone() as Intent)?.let {
                    if (it.hasRoute()) {
                        context.startActivity(it)
                        return
                    }
                }
                context.startActivity(this)
            }
        }
    }
}